synfig-studio/src/gui/mainwindow.cpp

```diff
@@ -98,26 +99,74 @@ MainWindow::MainWindow() :
 	bin_->add(*main_dock_book_);
 	bin_->show();
 
-	auto visible_vbox = manage(new Gtk::Box(Gtk::ORIENTATION_VERTICAL));
-	auto hidden_box   = manage(new Gtk::Box());
-
-	auto visible_menubar = App::ui_manager()->get_widget("/menubar-main");
-	auto hidden_menubar  = App::ui_manager()->get_widget("/menubar-hidden");
-	if (visible_menubar != NULL)
+	Glib::RefPtr<Gio::SimpleActionGroup> simple_action_group = Gio::SimpleActionGroup::create();
+	simple_action_group->add_action("menu-file");
+	simple_action_group->add_action("quit");
+
+	Glib::RefPtr<Gtk::Builder> builder;
+	builder = Gtk::Builder::create();
+
+	Glib::ustring ui_info = 
+		"<interface>"
+		"  <menu id='menubar'>"
+		"    <submenu>"
+		"      <attribute name='label' translatable='yes'>_File</attribute>"
+		"      <attribute name='action'>actions.menu-file</attribute>"
+		"      <section>"
+		"        <item>"
+		"          <attribute name='action'>actions.quit</attribute>"
+		"          <attribute name='label' translatable='yes'>_Quit</attribute>"
+		"        </item>"
+		"      </section>"
+		"    </submenu>"
+		"  </menu>"
+		"</interface>";
+
+	try
+	{
+		insert_action_group("actions",simple_action_group);
+		builder->add_from_string(ui_info);
+	}
+	catch(Glib::Error& error)
 	{
-		hidden_box->add(*hidden_menubar);
-		hidden_box->hide();
+		std::cerr << "Error while building the menus : " << error.what() << std::endl;
+	}
 
-		visible_vbox->add(*hidden_box);
-		visible_vbox->pack_start(*visible_menubar, false, false, 0);
+	auto builder_box = manage(new Gtk::Box(Gtk::ORIENTATION_VERTICAL));
```

Comment:
Why are you creating a Box?

Reply:
I added it in a extra box so that I can deal with that separate box only and have to deal with outside box as less as possible.

Reply:
I didn't understand

Reply:
I was wrong i guess. I removed that line now :)